taskana-adapter-camunda-tasklistener/src/main/java/pro/taskana/adapter/camunda/tasklistener/TaskanaTaskListener.java

```diff
@@ -34,153 +33,223 @@ public static TaskanaTaskListener getInstance() {
-                delegateTask.getName());
+            LOGGER.warn("Caught {} while trying to retrieve the " + propertyKey + " property from a process model", e);
 
         }
 
-        return domain;
+        return propertyValue;
 
     }
 
-    private String getClassification(DelegateTask delegateTask) {
 
-        String classification = null;
+    private String getUserTaskExtensionProperty(DelegateTask delegateTask, String propertyKey) {
+
+        String propertyValue = null;
 
         try {
-            CamundaProperties camundaProperties = delegateTask.getExecution()
-                .getBpmnModelElementInstance()
-                .getExtensionElements()
-                .getElementsQuery()
-                .filterByType(CamundaProperties.class)
-                .singleResult();
-
-            classification = camundaProperties.getCamundaProperties()
-                .stream()
-                .filter(camundaProperty -> camundaProperty.getCamundaName()
-                    .equals("classification"))
-                .collect(Collectors.toList())
-                .get(0)
-                .getCamundaValue();
 
+            ExtensionElements extensionElements = delegateTask.getExecution()
+                    .getBpmnModelElementInstance()
+                    .getExtensionElements();
+
+            if (extensionElements ==  null){
+                return propertyValue;
+            } else {
+                CamundaProperties camundaProperties = extensionElements.getElementsQuery()
+                        .filterByType(CamundaProperties.class).singleResult();
+
+                List<CamundaProperty> userTaskExtensionProperties = camundaProperties.getCamundaProperties()
+                        .stream()
+                        .filter(camundaProperty ->
+                                camundaProperty.getCamundaName()
+                                        .equals(propertyKey))
+                        .collect(Collectors.toList());
+
+                if (userTaskExtensionProperties.isEmpty()) {
+                    return propertyValue;
+                } else {
+                    propertyValue = userTaskExtensionProperties.get(0).getCamundaValue();
+                }
+            }
         } catch (Exception e) {
-            LOGGER.warn("Caught {} while trying to retrieve the classification of task {}", e, delegateTask.getName());
+            LOGGER.warn("Caught {} while trying to retrieve the " + propertyKey + " property of a user task", e);
         }
 
-        return classification;
+        return propertyValue;
     }
 
-    // TO-DO
-    private void insertDeleteEvent(DelegateTask delegateTask, Connection connection) {
+    private boolean taskWasCompletedByAdapter(DelegateTask delegateTask) {
 
-    }
+        String taskWasCompletedByAdapterValue = getUserTaskExtensionProperty(delegateTask, "taskWasCompletedByAdapter");
 
```

Comment:
An extension property cannot be used to find out whether the task was completed by the adapter nor not.
An extension property is defined in the model, therefore it has the same value for all process instances. However, if a task was completed due to an action of the adapter, or because an interrupting event fired or because the process instance was deleted is a event that can be different for each process instance. Therefore, we will need to introduce a runtime variable with task scope (remember, this flag could be different for each task inside the process) that must be set by the adapter and determines whether the adapter closed this task ...

Reply:
Indeed, this was old code and is deleted for now